### ISSUE TITLE
Get rid of warning ignored files

### ISSUE THREAD

**Issue by VadymBoguslavsky** [eslint/eslint#12249]
Hi, i have a question. After bump of sass-loader we need to use sassOptions in our next.config.js, but when i change it, like 
`  sassLoaderOptions: {
        sassOptions: {
          includePaths: ['public/assets/styles']
        }
}`

and try to commit to github, eslint gives me 
`0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override` 

Our default eslint is `yarn run eslint --fix --max-warnings=0`

Is there some way to ignore this warning?

**Comment by g-plane** [comment]
Hi @VadymBoguslavsky , can you show us your configuration?

**Comment by VadymBoguslavsky** [comment]
Ok. in `next.config.js` :

`const path = require('path');
const webpack = require('webpack')
const withSass = require('@zeit/next-sass');
const {
  WebpackBundleSizeAnalyzerPlugin
} = require('webpack-bundle-size-analyzer');
const { ANALYZE_SIZE } = process.env;


const withBundleAnalyzer = require("@next/bundle-analyzer")({
  enabled: process.env.ANALYZE === 'true',
});

const globalSass = [
  path.resolve(__dirname, 'styles/_mixins.scss'),
  path.resolve(__dirname, 'styles/_variables.scss'),
  path.resolve(__dirname, 'styles/_functions.scss'),
  path.resolve(__dirname, 'styles/_placeholders.scss')
]

module.exports = withBundleAnalyzer(withSass({
  sassLoaderOptions: {
    sassOptions: {
      includePaths: ['public/assets/styles']
    }
  },
  webpack: (config, { buildId, dev, isServer, defaultLoaders }) => {
    config.module.rules.push({
      enforce: "pre",
      test: /\.scss$/,
      loader: "sass-resources-loader",
      options: {
        resources: globalSass
      }
    });
    config.plugins.push(
      new webpack.IgnorePlugin(/^encoding$/, /node-fetch/)
    );
    if (ANALYZE_SIZE) {
      config.plugins.push(new WebpackBundleSizeAnalyzerPlugin('stats.txt'))
    }

    return config
  }
}));

module.exports.target = 'serverless'`

also we have .eslintignore where we adding next.config.js 

and part of package.json 

`  "husky": {
    "hooks": {
      "pre-commit": "lint-staged"
    }
  },
  "lint-staged": {
    "*.{js, jsx}": [
      "yarn run eslint --fix --max-warnings=0",
      "git add"
    ]
  }`

**Comment by g-plane** [comment]
What's your ESLint config?

**Comment by VadymBoguslavsky** [comment]
we have .eslintrc.yml 

---

```
plugins:
  - babel
  - react
  - jsx-a11y

env:
  node: true
  browser: true

parser: babel-eslint

parserOptions:
  ecmaFeatures:
    legacyDecorators: true

extends:
  - 'airbnb-base'
  - 'plugin:react/recommended'
  - 'plugin:jsx-a11y/recommended'
  - 'plugin:prettier/recommended'
  - 'prettier'

settings:
  react:
    version: 'latest'

rules:
  no-console: 0
  consistent-return: 0
  comma-dangle: 0
  jsx-a11y/label-has-associated-control: 0
  jsx-a11y/anchor-has-content: 0
  jsx-a11y/anchor-is-valid:
    [ 'error',
      { components: [ 'Link' ],
        aspects: [ 'invalidHref' ] } ]
  class-methods-use-this: [2, { "exceptMethods": ["componentDidMount"] }]

```


and ignore file .eslintignore
```

next.config.js
server.js

```

**Comment by aladdin-add** [comment]
seems you are trying to lint `next.config.js`, but it's been configured ingored somehow. I believe you just need to remove `"next.config.js"` in `.eslintignore`.

hope it helps!

**Comment by platinumazure** [comment]
@aladdin-add The challenge here is that the warning is unavoidable (since it doesn't come from a rule) and the user is using `--max-warnings`.

That said, I never understood from @VadymBoguslavsky why they have to run ESLint with that file ignored, and with a glob that picks up that file. Can either of those be changed?

**Comment by VadymBoguslavsky** [comment]
The problem is that we ignore our next.config.js using .eslintignore, and in theory it works, since when you remove next.config.js from .eslintignore, then errors are simply displayed. But I thought that if there is .eslintignore then everything that is in it should be ignored automatically, besides what is the point in this error? Oh and for now i remove  --max-warnings with --quiet.

**Comment by adhearn** [comment]
We're having a similar problem here. We have a git hook that lints and will fail if there are any warnings, and configured our .eslintignore to ignore files that we didn't want to be warned about. However, we're now getting a warning about there being issues in a file that we explicitly say to ignore. This behavior is at best confusing, and at worst totally incorrect.

**Comment by kaicataldo** [comment]
@VadymBoguslavsky Do you mind clarifying exactly what command you're running ESLint with to get the results you've reported above? It initially sounded like the error was about running ESLint on files that are ignored, but now it's sounding like there are other errors it's reporting in the file?

@adhearn I think your issue is different, and it sounds like a duplicate of https://github.com/eslint/eslint/issues/12206.

**Comment by VadymBoguslavsky** [comment]
@kaicataldo The command is `yarn run eslint --fix --max-warnings=0` but for now i changed it to `--quiet` to get rid of the warning above. And yeah we have errors in next.config.js but thats why we put it in .eslintignore

**Comment by kaicataldo** [comment]
Sorry, I'm still unsure how it gets in this state. Since you aren't passing file paths directly to the CLI, it shouldn't be trying to lint any files in `.eslintignore`. That being said, there are a lot of moving parts here and maybe I'm just missing something. Any chance you could create a GitHub repository with a minimal reproduction case so we can better help you?

**Comment by tnrich** [comment]
Hey there, sorry to hijack this thread but I'm dealing with a similar issue: 

Here's the situation:
Due to changes in v9, lint-staged now passes a list of all changed files to eslint (instead of a configurable subset), some of which I do not want to be linted (hence I ignore them with my .eslintignore). Eslint throws warnings that I am trying to lint ignored files. 

Is there a way for eslint to suppress warnings that files are being ignored?

Do we need to add another CLI option to get around this? Does one already exist? 

Here is the original issue in lint-staged: https://github.com/okonet/lint-staged/issues/711

**Comment by VadymBoguslavsky** [comment]
Just change your `--max-warnings=0` to `--quiet`. It will suppress all the warnings you have and show only errors.
As i understood, there is no other way to fix it.

**Comment by tnrich** [comment]
@VadymBoguslavsky that won't work for me unfortunately. Luckily it looks like there has been an RFC put in to be able to avoid this behavior: https://github.com/eslint/eslint/issues/12206

**Comment by puresick** [comment]
I am having a similar issue where I need to surpress the warning of ignored files.
Anything new on this issue yet?
